windows/src/developer/samples/imsample/imsample.cpp

```diff
@@ -426,10 +432,19 @@ LRESULT CALLBACK IMSampleChildWndProc(HWND hwndChild, UINT msg, WPARAM wParam, L
 
 		return 0;
 	case WM_PAINT:
+    if (!curkbd) {
+       return 0;
+    }
+    if (!curkbd->hFont) {
+       return 0;
+    }
+    if (!curkbd->groups) {
+      return 0;
+    }
 		if(!PrepIM()) return DefWindowProc(hwndChild, msg, wParam, lParam);
```

Comment:
```suggestion
    if (!curkbd || !curkbd->hFont || !curkbd->groups || !PrepIM()) {
      return DefWindowProc(hwndChild, msg, wParam, lParam);
    }
```

You need to call `BeginPaint`/`EndPaint` on a `WM_PAINT`, or else `return DefWindowProc()` in order to validate the paint region, otherwise `WM_PAINT` will keep getting called.